Packs/DuoAdminApi/Integrations/DuoEventCollector/DuoEventCollector.py

```diff
@@ -93,48 +157,56 @@ def rotate_request_order(self) -> None:
         temp.rotate(-1)
         self.request_order = list(temp)
 
-    def make_sdk_call(self):  # pragma: no cover
-        events: list = self.client.call(self.request_order)  # type: ignore
-        events = sorted(events, key=lambda e: e['timestamp'])
+    def make_sdk_call(self) -> tuple:
+        events, metadata = self.client.call(self.request_order)
         events = events[: int(self.client.params.limit)]
-        return events
+        return events, metadata
 
-    def _iter_events(self) -> None:  # type: ignore  # pragma: no cover
+    def _iter_events(self) -> Generator:
         """
         Function that responsible for the iteration over the events returned from the Duo api
         """
-        events: list = self.make_sdk_call()
+        events, metadata = self.make_sdk_call()
         while True:
             if events:
-                self.client.set_next_run_filter(events[-1]['timestamp'], self.request_order[0])
+                if self.request_order[0] in [
+                    LogType.ADMINISTRATION,
+                    LogType.TELEPHONY,
+                ]:
+                    self.client.set_next_run_filter_v1(self.request_order[0], events[-1]['timestamp'])
+                else:
+                    self.client.set_next_run_filter_v2(self.request_order[0], metadata)
```

Comment:
Can you please add a comment explaining why there's a different next run filter for v1 vs v2 (will this need to change in the future?)

Reply:
Done